pandas/core/generic.py

```diff
@@ -2501,12 +2501,15 @@ def to_sql(
         schema : str, optional
             Specify the schema (if database flavor supports this). If None, use
             default schema.
-        if_exists : {'fail', 'replace', 'append'}, default 'fail'
+        if_exists : {'fail', 'replace', 'append', 'upsert_overwrite', 'upsert_keep'},\
+        default 'fail'
```

Comment:
this api is conflating two things. isn't upsert by-definition to overwrite? what is the usecase for 'upsert_keep'?

Reply:
Over the course of this PR I have come to agree that using the term `upsert`, in this case is not clear.  In reality this behaviour is more akin to an `on_conflict` argument (of which I suppose upsert is a subset).  Do you think it'd be cleaner to add an `on_conflict` argument to the `to_sql` method signature, which takes parameters `do nothing` or `overwrite`?

Reply:
I think that makes sense. `overwrite` could also be called `do update`.